histogram.go

```
@@ -368,7 +368,7 @@ func ExponentialDurationBuckets(start time.Duration, factor float64, n int) (Dur
 	curr := start
 	for i := range buckets {
 		buckets[i] = curr
-		curr = time.Duration(float64(curr) * factor)
+		curr = time.Duration(safeFloat64ToInt64(float64(curr) * factor))
```

this can end up generating invalid buckets. e.g. if you're generating 10 buckets and bucket 7 is already overflowing, the last 3 buckets are going to be redundant/duplicates; this will cause issues on the metric consumption side (`histogramPercentile` in m3ql will complain about duplicate bucket ranges) 

i.e. we should probably pick a semantic in terms of either truncating to generate only valid, non-duplicate buckets OR we should consider raising a panic when there's invalid bucket definitions. not convinced which is better - but we should pick a path in this diff.

---

I thought about removing duplicate buckets, but my concern is that the third argument of the `ExponentialDurationBuckets` function is the number of buckets so that some clients may rely on that number.
Also, raising a panic is not a backward-compatible change.